065-topic-replication-factor-change.md

```diff
@@ -0,0 +1,292 @@
+This is configurable for all endpoints in `Kafka.spec.cruiseControl.config` using the `default.replication.throttle` property.
+
+### User tasks
+
+The [`user_tasks`](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-user-request-result) endpoint will be used to periodically check the state of requested tasks.
+The only required parameter is the `User-Task-ID`, which is returned by the `topic_configuration` response.
+
+```sh
+$ curl -s "localhost:9090/kafkacruisecontrol/user_tasks?user_task_ids=5344ca89-351f-4565-8d0f-9aade00e053d&json=true" | jq
+{
+  "userTasks": [
+    {
+      "Status": "Completed",
+      "ClientIdentity": "127.0.0.1",
+      "RequestURL": "POST /kafkacruisecontrol/topic_configuration?topic=my-topic&replication_factor=2&skip_rack_awareness_check=true&dryrun=false&json=true",
+      "UserTaskId": "5344ca89-351f-4565-8d0f-9aade00e053d",
+      "StartMs": "1701961135978"
+    }
+  ],
+  "version": 1
+}
+```
+
+### Configuration parameters
+
+The Topic Operator will need to authenticate and send HTTP requests to the Cruise Control's REST API.
+A new admin user will be provisioned for the Topic Operator, separate from the credentials used by the KafkaRebalanceOperator.
+The following environment variables will be set by the Cluster Operator for consumption by the Topic Operator when Cruise Control is configured in Kafka resource, or manually by the user in case of standalone deployment:
+
+| Name                                  | Type    | Default | Description                                                   |
+|:--------------------------------------|---------|---------|---------------------------------------------------------------|
+| STRIMZI_CRUISE_CONTROL_ENABLED        | Boolean | false   | Whether Cruise Control integration is enabled                 |
+| STRIMZI_CRUISE_CONTROL_RACK_ENABLED   | Boolean | false   | Whether rack awareness is enabled in the target Kafka cluster |
+| STRIMZI_CRUISE_CONTROL_HOSTNAME       | String  | ""      | Cruise Control hostname                                       |
+| STRIMZI_CRUISE_CONTROL_PORT           | Integer | 9090    | Cruise Control port                                           |
+| STRIMZI_CRUISE_CONTROL_SSL_ENABLED    | Boolean | false   | Whether SSL encryption is enabled                             |
+| STRIMZI_CRUISE_CONTROL_AUTH_ENABLED   | Boolean | false   | Whether Basic authentication is enabled                       |
+
+With the Cluster Operator deployment, the certificate chain in PEM format and the admin user credentials for the Cruise Control REST API will be automatically mounted in the Topic Operator container.
+Instead, with the standalone deployment, the user will have to mount: certificate chain in `/etc/tls-sidecar/cluster-ca-certs/ca.crt`, credentials in `/etc/eto-cc-api/cruise-control.apiToAdminUsername` and `/etc/eto-cc-api/cruise-control.apiToAdminPassword`.
```

Comment:
What does `apiToAdminUsername` / `apiToAdminPassword` stand for? Isn't it better to name it some more expected way? `cruise-control-api.username` / `cruise-control-api.password` for example? Or what does `apiToAdmin` mean? It sounds like it is some API to communicate with some Administrator ... but that is not right. It is admin credentials to communicate with an API. So if anything, than `AdminUsernameForApi`.

Also, why not mount these as environment variables instead of files? Wouldn't that make it easier to work with?

Reply:
> What does apiToAdminUsername / apiToAdminPassword stand for?

Topic operator (To). I know it's not great, so I'm open to naming suggestions that make it clear that they are used by the TO.

> why not mount these as environment variables instead of files? Wouldn't that make it easier to work with

I want to maintain consistency with certificate handling.

Reply:
> Topic operator (To). I know it's not great, so I'm open to naming suggestions that make it clear that they are used by the TO.

Well, we normally use Topic **O**perator => so T**O**. I think the `To` is really confusing given it has also plain English meaning - it honestly never ever came to my mind that this is supposed to mean Topic Operator and I'm using the shortcut TO daily. I would call it `cruise-control-api.username` and `cruise-control-api.password`. But if you wanna keep it, you should definitely use `TO`.

Reply:
> I want to maintain consistency with certificate handling.

It is normally considered less secure with file and there seems no reason for it. I would also expect that one day the certificate handling will be reworked. But ok, if you wanna keep it its fine, I can live with it.